api/v1alpha1/operator_types_test.go

```diff
@@ -8,39 +8,51 @@ import (
 	"io/fs"
 	"strconv"
 	"strings"
-
-	. "github.com/onsi/ginkgo/v2"
-	. "github.com/onsi/gomega"
+	"testing"
 
 	"github.com/operator-framework/operator-controller/internal/conditionsets"
+	// replaces with "slices" in go 1.21
+	"golang.org/x/exp/slices"
 )
 
-var _ = Describe("OperatorTypes", func() {
-	Describe("Condition Type and Reason constants", func() {
-		It("should register types in conditionsets.ConditionTypes", func() {
-			types, err := parseConstants("Type")
-			Expect(err).NotTo(HaveOccurred())
+func TestOperatorType(t *testing.T) {
```

Comment:
Nit: better test name to convey test purpose a bit better. Same comment for the Reason tests.
```suggestion
func TestOperatorTypeRegistration(t *testing.T) {
```